crates/sui-indexer-alt-framework/src/handlers/cp_sequence_numbers.rs

```diff
@@ -22,6 +23,80 @@ pub struct StoredCpSequenceNumbers {
 
 pub struct CpSequenceNumbers;
 
+/// A struct that can be instantiated by the pruner task to map a `from` and `to` checkpoint to its
+/// corresponding `tx_lo` and containing epoch. The `from` checkpoint is expected to be inclusive,
+/// and the `to` checkpoint is exclusive. This requires the existence of the `checkpoint_metadata`
+/// table.
+pub struct PrunableRange {
+    from: StoredCpSequenceNumbers,
+    to: StoredCpSequenceNumbers,
+}
+
+#[derive(Debug, thiserror::Error)]
+pub enum RangeError {
+    #[error("Invalid checkpoint range: `from` {0} is greater than `to` {1}")]
+    InvalidCheckpointRange(u64, u64),
+    #[error("No checkpoint mapping found for checkpoints {0} and {1}")]
+    NoCheckpointMapping(u64, u64),
+    #[error("Database error: {0}")]
+    DbError(diesel::result::Error),
+}
+
+impl PrunableRange {
```

Comment:
AFAICT, each pipeline would need to either fetch a transaction interval, or an epoch interval -- we no longer need to provide all three. Would it be better to provide two functions -- one that translates `from_cp, to_cp` into a tx_interval, and another into an `epoch` interval? You can re-use Rust's [`Range`](https://doc.rust-lang.org/std/ops/struct.Range.html) type to make the API clear (about the ends being exclusive:

```rust
async fn tx_interval(conn: &mut Connection<'_>, cps: Range<u64>) -> Range<u64>;
async fn epoch_interval(conn: &mut Connection<'_>, cps: Range<u64>) -> Range<u64>;
```